### PR TITLE
fix: key assignment genesis tests

### PR DESCRIPTION
**Pull request by sainoe**
* Fix broken expectations in consumer genesis tests
* Update provider genesis tests

### CODE REVIEW

**Review by mpoke — APPROVED** [review]
LGTM

**Comment by mpoke** [comment]
@sainoe Why are the test still failing?

File: x/ccv/provider/keeper/msg_server.go [hunk 1]
```diff
@@ -35,16 +35,24 @@ func (k msgServer) AssignConsumerKey(goCtx context.Context, msg *types.MsgAssign
 	// Note that current attack potential is restricted because validators must sign
 	// the transaction, and the chainID size is limited.
 
-	providerAddr, err := sdk.ConsAddressFromBech32(msg.ProviderAddr)
+	providerValidatorAddr, err := sdk.ValAddressFromBech32(msg.ProviderAddr)
```
Comment (mpoke) [inline x/ccv/provider/keeper/msg_server.go:38]: see https://github.com/cosmos/interchain-security/pull/515#discussion_r1031700921

**Review by mpoke — CHANGES_REQUESTED** [review]

File: x/ccv/provider/keeper/msg_server.go (hunk 1 above)
Comment (sainoe) [reply x/ccv/provider/keeper/msg_server.go:38]: see https://github.com/cosmos/interchain-security/pull/515#discussion_r1031738462

**Review by mpoke — APPROVED** [review]

**Pull request merged**